src/planwise/client/scenarios/views.cljs

```diff
@@ -16,16 +16,23 @@
             [planwise.client.utils :as utils]
             [planwise.client.ui.rmwc :as m]))
 
+(defn- show-provider
+  [{{:keys [action name capacity investment]} :elem :as provider}]
+  (if (nil? action)
+    (str "<b>" name "</b><br> Capacity: " capacity)
+    (str "<b> New provider " (:index provider) "</b><br> Click on panel for editing... ")))
+
 (defn simple-map
-  [{:keys [bbox]} {:keys [changeset raster]}]
+  [{:keys [bbox]} scenario]
   (let [state    (subscribe [:scenarios/view-state])
         index    (subscribe [:scenarios/changeset-index])
         position (r/atom mapping/map-preview-position)
         zoom     (r/atom 3)
         add-point (fn [lat lon] (dispatch [:scenarios/create-provider {:lat lat
                                                                        :lon lon}]))]
-    (fn [{:keys [bbox]} {:keys [changeset raster]}]
-      (let [indexed-changeset     (map (fn [elem] {:elem elem :index (.indexOf changeset elem)}) changeset)
+    (fn [{:keys [bbox]} {:keys [changeset providers raster] :as scenario}]
+      (let [changeset   (into providers changeset)
+            indexed-changeset     (map (fn [elem] {:elem elem :index (.indexOf changeset elem)}) changeset)
```

Comment:
Pretty sure `.indexOf` doesn't work here, since `changeset` should be a Clojurescript structure and not a Javascript array. You can use `map-indexed` to achieve similar results.